src/pools/cluster/dynamic.ts

```diff
@@ -47,31 +55,12 @@ export class DynamicClusterPool<
    * @returns Cluster worker.
    */
   protected chooseWorker (): Worker {
-    for (const [worker, numberOfTasks] of this.tasks) {
-      if (numberOfTasks === 0) {
-        // A worker is free, use it
-        return worker
-      }
-    }
-
-    if (this.workers.length === this.max) {
-      this.emitter.emit('FullPool')
-      return super.chooseWorker()
-    }
-
-    // All workers are busy, create a new worker
-    const workerCreated = this.createAndSetupWorker()
-    this.registerWorkerMessageListener<Data>(workerCreated, message => {
-      const tasksInProgress = this.tasks.get(workerCreated)
-      if (
-        isKillBehavior(KillBehaviors.HARD, message.kill) ||
-        tasksInProgress === 0
-      ) {
-        // Kill received from the worker, means that no new tasks are submitted to that worker for a while ( > maxInactiveTime)
-        this.sendToWorker(workerCreated, { kill: 1 })
-        void this.destroyWorker(workerCreated)
-      }
-    })
-    return workerCreated
+    return dynamicallyChooseWorker(
```

Comment:
I'm not very comfortable with a method returning another method call ... 
I would expect here to be able to register the callback that will be used to choose a worker: 

+ fixed -> register the function roundRobinSelection
+ dynamic -> register the function dynamicallyChooseWorker

Reply:
@jerome-benoit  yes this is something like my proposal, having a property ( i.e. isDynamic ) and based on that property take actions, so we can even move stuff to the abstract class

Reply:
Problem about this is that the dynamic choose worker function does so many (3) things that rely on stuff within the DynamicPool class itself that it cannot be just a function that can be swapped.

The fixed pool already do the round robin basically because it uses the default that is defined by/in AbstractPool

I think I know what your thoughts are. Just doing like

```ts
class DynamicCPool ... {
 chooseStrategy = (args) => Worker
}

class FixedCPool ... {
 chooseStrategy = (args) => Worker
}
```

but you are loosing the whole type-safety I think :thinking:

Reply:
> ```ts
> class DynamicCPool ... {
>  chooseStrategy = (args) => Worker
> }
> 
> class FixedCPool ... {
>  chooseStrategy = (args) => Worker
> }
> ```

I think that fine to use `...args: any[]`. 
You can't have a one size fits all here regarding types for algo that requires very different inputs.

Reply:
Also, if you can define the function to use for worker choice and register it, you can then use arrow function and they will automatically be binded to this.